### DISCUSSION TITLE
How to respond to memory pressure events for LocalLLM?

### DISCUSSION THREAD

**Discussion by bryan1anderson** [StanfordSpezi/.github#69] [Challenges and Help] [answered]
### What Stanford Spezi module is your challenge related to?

Spezi

### Description

I get them pretty regularly on iOS. I need a way to respond to them, deinit everything in cache and then load them back up when appropriate

### Reproduction

                                    let schema = LLMLocalSchema(
                                        model: llmLocalModel,
                                        parameters: .init(
                                            systemPrompt:
                                                "system prompt",
                                            displayEveryNTokens: 1
                                        ),
                                        samplingParameters: .init(temperature: 0.1),
                                        injectIntoContext: true
                                    )

                                    let session = runner(with: schema)

session.setup()

### Expected behavior

The ability to clear memory to avoid getting killed

### Code of Conduct

- [X] I agree to follow this projects's Code of Conduct and Contributing Guidelines

**Comment by philippzagar** [comment, marked as answer]
Thanks @bryan1anderson for the great post, this is definitely a required feature of SpeziLLM in real-world use scenarios! 🚀 

Formalized as a feature request in https://github.com/StanfordSpezi/SpeziLLM/issues/90
